### PR TITLE
Luisae/highlight tagging

### PR DESCRIPTION
**Pull request by LuisaE**
Add highlight tagging and method to change between highlighting methods

### CODE REVIEW

**Comment by sukritsangvong** [comment]
@etiennerichart can continue working on this branch as a test branch. (may be a good idea to merge main to this branch first after #50 is merged)

**Comment by LuisaE** [comment]
@etiennerichart in your last commit I see a bunch of good stuff that should be merged and some experimental stuff that shouldn't. Do you want to separate this out and make a PR with the stuff we want to keep? Should we meet to discuss?

**Review by RichartE — APPROVED** [review]

File: wikichange/src/content/markContent.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        });
+        return { succeed, fail };
+    };
+    return markContent;
+}
+
+/**
+ * Get the markContent function created depending on the HIGHLIGHT_TYPE
+ */
+const markContent = (() => {
```
Comment (sukritsangvong) [inline wikichange/src/content/markContent.js:59]: should change them to have different names

File: wikichange/src/content/tagEveryWord.js [hunk 2]
```diff
@@ -12,9 +12,9 @@ const tagEveryWord = () => {
                 return NodeFilter.FILTER_REJECT;
             }
             // Skip the references
-            if (node.parentElement.closest('.reflist')){
-                return NodeFilter.FILTER_REJECT;
-            }
+            // if (node.parentElement.closest('.reflist')){
+            //     return NodeFilter.FILTER_REJECT;
+            // }
```
Comment (sukritsangvong) [inline wikichange/src/content/tagEveryWord.js:17]: do we still need this?

**Review by sukritsangvong — CHANGES_REQUESTED** [review]
2 minor fixes

**Comment by sukritsangvong** [comment]
Do I understand correctly that this is still highlighting the same amount as the node version, but this makes it easier to improve the algorithm later as we can access all the texts in the page instead of iterating through DOM tree?

File: wikichange/src/content/tagEveryWord.js (hunk 2 above)
Comment (LuisaE) [reply wikichange/src/content/tagEveryWord.js:17]: Yes, @etiennerichart told me we should keep this in case of heavier pages like US

**Comment by LuisaE** [comment]
> Do I understand correctly that this is still highlighting the same amount as the node version, but this makes it easier to improve the algorithm later as we can access all the texts in the page instead of iterating through DOM tree?

Yes, this is what I understood from it. Actually highlighting a little less right now as doesn't do anything to the start of paragraphs

**Review by sukritsangvong — APPROVED** [review]

**Pull request merged**